File: 프로그래머스/2/42586. 기능개발/기능개발.py
def solution(progresses, speeds):
    from math import ceil
    answer = []

    days=[ceil((100-p)/s) for p,s in zip(progresses, speeds)]
    
    while days:
        current_day=days.pop(0)
        count=1
        
        while days and days[0]<=current_day:
            days.pop(0)
            count+=1
            
        answer.append(count)
    
    return answer

File: 프로그래머스/2/42586. 기능개발/기능개발.js
function solution(progresses, speeds) {
  const remainingDays = progresses.map((progress, index) =>
    Math.ceil((100 - progress) / speeds[index])
  );

  const result = [];
  let maxDay = remainingDays[0];
  let count = 0;

  for (let day of remainingDays) {
    if (day <= maxDay) {
      count++;
    } else {
      result.push(count);
      count = 1;
      maxDay = day;
    }
  }

  result.push(count);
  return result;
}
